motor_control/Triple_HBridge/ramping.h

```diff
@@ -0,0 +1,42 @@
+#ifndef RAMPING_H
+#define RAMPING_H
+
+class Ramping{
+public:
+  Ramping();
+  virtual byte newSpd(int throttle) = 0;
+  
+protected:
+  void setThrottle(int throttle);
+  void setCurrentSpd(byte spd); 
+  void setDesiredSpd(byte spd);
+
+  int  getThrottle()   const;
+  byte getCurrentSpd() const;
+  byte getDesiredSpd() const;
+
+  void spdUp(byte amount);
+  void spdDown(byte amount);
+private:
+  int  m_throttleIP;
+  byte m_currentSpd;
+  byte m_desiredSpd;
+};
+
+class Linear : public Ramping{
+public:
+  Linear(byte rampSpd);
+  byte newSpd(int throttle);
+private:
+  byte m_RAMPSPD;
+};
+
+class PD : public Ramping{
```

Comment:
What does PD stand for?

Reply:
It's Proportional and Derivative control, but this implementation doesn't account time yet